subsys/usb/usb_c/usbc_tc_common.c

```diff
@@ -215,14 +215,23 @@ static void tc_cc_open_entry(void *obj)
 	const struct device *dev = tc->dev;
 	struct usbc_port_data *data = dev->data;
 	const struct device *tcpc = data->tcpc;
+	int ret;
 
 	tc->cc_voltage = TC_CC_VOLT_OPEN;
 
 	/* Disable VCONN */
-	tcpc_set_vconn(tcpc, false);
+	ret = tcpc_set_vconn(tcpc, false);
```

Comment:
We should think about what to do if the tcpc_set_* functions fail. Maybe log the error as you're doing and then go to the error recovery state, but I'm not sure? Let's discuss offline.

Reply:
We should set the state to error recovery in both states

Reply:
As we discussed, the TCPC init failure will disable the type-c layer, but other failures during the runtime will move to the error recovery state.